packages/plugin-ext/src/main/node/plugins-key-value-storage.ts

```diff
@@ -110,7 +110,16 @@ export class PluginsKeyValueStorage {
 
     private async writeToFile(pathToFile: string, data: KeysToKeysToAnyValue): Promise<void> {
         await fs.ensureDir(path.dirname(pathToFile));
-        await fs.writeJSON(pathToFile, data);
+        const tmpFile = pathToFile + '.tmp';
+        await new Promise((resolve, reject) => {
```

Comment:
the solution looks strange to me. There should be a way to call only one fs mehtod.

Reply:
I've tried already a few options, and this way in fact does not also. It fails with the error, .tmp files does not exist.
`fs.promises`, does not help as well.

The main issue actually not here, but in subsequent `read`, and it happens when OS did not finish an actual write, and the file is completely empty. I think the best way would be to use write to disk only as a way to keep file up to date, but read it only if have not changed since our last read. And even do not write if we have not changed anything.